In [1]:
import pandas as pd
import numpy as np
from scipy.stats import norm
from scipy.stats import t
import math

/Users/zihanding/opt/anaconda3/lib/python3.9/site-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.24.2
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"


In [4]:
df = pd.read_csv("/Users/zihanding/Developer/CSE544_Project/3.2_linear_interpolation.csv")
# df['Year'] = [df['DATE'][i][0:4] for i in range(len(df))]

In [5]:
df

DATE  CPI_all_items  Rent_of_Primary_residence  \
0    1997-01-01         159.40                    179.400   
1    1997-02-01         159.70                    180.100   
2    1997-03-01         159.80                    180.300   
3    1997-04-01         159.90                    180.700   
4    1997-05-01         159.90                    181.400   
..          ...            ...                        ...   
307  2022-08-01         295.32                    377.091   
308  2022-09-01         295.32                    379.085   
309  2022-10-01         295.32                    383.055   
310  2022-11-01         295.32                    384.450   
311  2022-12-01         295.32                    388.585   

     Monthly_Housing_Cost  CPI_Energy  US_Dollar_Purchasing_power  
0              160.400000     155.300                        62.8  
1              161.100000     155.600                        62.6  
2              161.000000     156.000                        62.5  
3              160.900000     156.200                        62.4  
4              161.100000     156.300                        62.5  
..                    ...         ...                         ...  
307            274.419286     334.212                        33.8  
308            274.458714     337.369                        33.7  
309            274.498143     340.532                        33.6  
310            274.537571     342.266                        33.6  
311            274.577000     342.266                        33.7  

[312 rows x 6 columns]

In [17]:
house = np.array(df['Monthly_Housing_Cost'])
m_house = np.median(house)
all_item = np.array(df['CPI_all_items'])
m_all = np.median(all_item)

In [18]:
df['House'] = ['High' if df['Monthly_Housing_Cost'][i] > m_house else 'Low' for i in range(len(df))]

In [26]:
df['All'] = ['High' if df['Monthly_Housing_Cost'][i] > m_all else 'Low' for i in range(len(df))]

In [27]:
df

DATE  CPI_all_items  Rent_of_Primary_residence  \
0    1997-01-01         159.40                    179.400   
1    1997-02-01         159.70                    180.100   
2    1997-03-01         159.80                    180.300   
3    1997-04-01         159.90                    180.700   
4    1997-05-01         159.90                    181.400   
..          ...            ...                        ...   
307  2022-08-01         295.32                    377.091   
308  2022-09-01         295.32                    379.085   
309  2022-10-01         295.32                    383.055   
310  2022-11-01         295.32                    384.450   
311  2022-12-01         295.32                    388.585   

     Monthly_Housing_Cost  CPI_Energy  US_Dollar_Purchasing_power House   All  
0              160.400000     155.300                        62.8   Low   Low  
1              161.100000     155.600                        62.6   Low   Low  
2              161.000000     156.000                        62.5   Low   Low  
3              160.900000     156.200                        62.4   Low   Low  
4              161.100000     156.300                        62.5   Low   Low  
..                    ...         ...                         ...   ...   ...  
307            274.419286     334.212                        33.8  High  High  
308            274.458714     337.369                        33.7  High  High  
309            274.498143     340.532                        33.6  High  High  
310            274.537571     342.266                        33.6  High  High  
311            274.577000     342.266                        33.7  High  High  

[312 rows x 8 columns]

In [28]:
condition = (df['House']=='High') & (df['All'] =='High')
result = df[condition]
print(len(result))

condition = (df['House']=='High') & (df['All'] =='Low')
result = df[condition]
print(len(result))

condition = (df['House']=='Low') & (df['All'] =='High')
result = df[condition]
print(len(result))

condition = (df['House']=='Low') & (df['All'] =='Low')
result = df[condition]
print(len(result))

143
13
0
156


The pr of each col is 1/2,1/2

In [32]:
Qobs = ((71.5-143)**2+71.5**2)/71.5 + ((84.5-13)**2+(84.5-156)**2)/84.5
print(Qobs)

264.0


In [33]:
from scipy.stats import chi2_contingency

Qobserved = np.array([[143,0],[13,156]])
chi2, p, dof, expected = chi2_contingency(Qobserved)
print(chi2,p,dof,expected)

260.32060247444866 1.4609408098142176e-58 1 [[71.5 71.5]
 [84.5 84.5]]


H0: Housing and All item is not related.  
H1: Housing and All item are related.  

In [34]:
p>0.05

False

| |High Housing Cost | Low Housing Cost | Total |
| :----:| :----: | :----: |:----: |
| High All Items Cost | (O:143)(E:71.5) | (O:0)(E:71.5) |143|
| Low All Items Cost | (O:13)(E:84.5) | (O:156)(E:84.5) |169|
| Total | 156 | 156 |312|